File: src/main/java/org/goafabric/catalog/controller/InsuranceController.java
package org.goafabric.catalog.controller;

import org.goafabric.catalog.job.insurance.InsuranceEo;
import org.goafabric.catalog.job.insurance.InsuranceRepository;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping(value = "insurances", produces = MediaType.APPLICATION_JSON_VALUE)
public class InsuranceController {
    private final InsuranceRepository repository;

    public InsuranceController(InsuranceRepository repository) {
        this.repository = repository;
    }

    @GetMapping("/{id}")
    public InsuranceEo getById(@PathVariable String id) {
        return repository.findById(id).orElseThrow();
    }

    @GetMapping("/findByCode")
    public List<InsuranceEo> findByCode(@RequestParam("code") String code) {
        return repository.findByCode(code);
    }

    @GetMapping("/findByDisplay")
    public List<InsuranceEo> findByDisplay(@RequestParam("display") String display) {
        return repository.findByDisplayStartsWith(display);
    }
}

File: src/main/kotlin/org/goafabric/catalog/controller/InsuranceController.kt
package org.goafabric.catalog.controller

import org.goafabric.catalog.job.insurance.InsuranceEo
import org.goafabric.catalog.job.insurance.InsuranceRepository
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping(value = ["insurances"], produces = [MediaType.APPLICATION_JSON_VALUE])
class InsuranceController(private val repository: InsuranceRepository) {
    @GetMapping("/{id}")
    fun getById(@PathVariable id: String): InsuranceEo {
        return repository.findById(id).orElseThrow()!!
    }

    @GetMapping("/findByCode")
    fun findByCode(@RequestParam("code") code: String): MutableList<InsuranceEo> {
        return repository.findByCode(code)
    }

    @GetMapping("/findByDisplay")
    fun findByDisplay(@RequestParam("display") display: String): MutableList<InsuranceEo> {
        return repository.findByDisplayStartsWith(display)
    }
}
